AsyncDisplayKit/Private/ASEnvironmentInternal.mm

```diff
@@ -15,11 +15,17 @@
 //#define LOG(...) NSLog(__VA_ARGS__)
 #define LOG(...)
 
-#define AS_SUPPORT_PROPAGATION NO
+#define AS_SUPPORT_UPWARD_PROPAGATION NO
+#define AS_SUPPORT_DOWNWARD_PROPAGATION YES
 
-BOOL ASEnvironmentStatePropagationEnabled()
+BOOL ASEnvironmentStateUpwardPropagationEnabled()
 {
-  return AS_SUPPORT_PROPAGATION;
+  return AS_SUPPORT_UPWARD_PROPAGATION;
+}
+
+BOOL ASEnvironmentStateDownwardPropagationEnabled()
+{
+  return AS_SUPPORT_DOWNWARD_PROPAGATION;
 }
```

Comment:
As we are currently prevent certain state to propagate should we be more discrete what we support for now? Something like -supportsTraitsCollectionPropagation and ASEnvironmentStateTraitCollectionPropagationEnabled? Maybe we can add that to special header file that if we finally decide to support upward and downward propagation for all states we can just remove it?